core/src/main/java/org/elasticsearch/cli/EnvironmentAwareCommand.java

```diff
@@ -59,17 +67,22 @@ protected void execute(Terminal terminal, OptionSet options) throws Exception {
             settings.put(kvp.key, kvp.value);
         }
 
-        putSystemPropertyIfSettingIsMissing(settings, "path.conf", "es.path.conf");
         putSystemPropertyIfSettingIsMissing(settings, "path.data", "es.path.data");
         putSystemPropertyIfSettingIsMissing(settings, "path.home", "es.path.home");
         putSystemPropertyIfSettingIsMissing(settings, "path.logs", "es.path.logs");
 
-        execute(terminal, options, createEnv(terminal, settings));
+        final String pathConf = pathConfOption.value(options);
+        execute(terminal, options, createEnv(terminal, settings, getPathConf(pathConf)));
+    }
+
+    @SuppressForbidden(reason = "need path to construct environment")
+    private static Path getPathConf(final String pathConf) {
```

Comment:
I know this setting name was (and argument now is) `path.conf`, but can we call this getConfPath?  It doesn't really make sense without the dot (might even call it getConfigPath).